sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -398,133 +384,117 @@ export class EventHubSender extends LinkEntity {
-            this._context.connectionId,
-            this.name
-          );
-          if (timeoutInMs <= timeTakenByInit) {
-            actionAfterTimeout();
-            return;
-          }
-          try {
-            this._sender!.sendTimeoutInSeconds = (timeoutInMs - timeTakenByInit) / 1000;
-            const delivery = await this._sender!.send(rheaMessage, undefined, 0x80013700);
-            logger.info(
-              "[%s] Sender '%s', sent message with delivery id: %d",
-              this._context.connectionId,
-              this.name,
-              delivery.id
-            );
-            return resolve();
-          } catch (err) {
-            const translatedError = translate(err.innerError || err);
-            logger.warning(
-              "[%s] An error occurred while sending the message %s",
-              this._context.connectionId,
-              `${translatedError?.name}: ${translatedError?.message}`
-            );
-            logErrorStackTrace(translatedError);
-            return reject(translatedError);
-          } finally {
-            removeListeners();
-          }
-        } else {
-          // let us retry to send the message after some time.
-          const msg =
-            `[${this._context.connectionId}] Sender "${this.name}", ` +
-            `cannot send the message right now. Please try later.`;
-          logger.warning(msg);
-          const amqpError: AmqpError = {
-            condition: ErrorNameConditionMapper.SenderBusyError,
-            description: msg
-          };
-          reject(translate(amqpError));
-        }
-      });
+      }
+
+      if (!this._sender!.sendable()) {
+        // let us retry to send the message after some time.
+        const msg =
+          `[${this._context.connectionId}] Sender "${this.name}", ` +
+          `cannot send the message right now. Please try later.`;
+        logger.warning(msg);
+        const amqpError: AmqpError = {
+          condition: ErrorNameConditionMapper.SenderBusyError,
+          description: msg
+        };
+        throw translate(amqpError);
+      }
+
+      logger.verbose(
+        "[%s] Sender '%s', sending message with id '%s'.",
+        this._context.connectionId,
+        this.name
+      );
+      if (timeoutInMs <= timeTakenByInit) {
+        const desc: string =
+          `${this._context.connectionId} Sender "${this.name}" ` +
+          `with address "${this.address}", was not able to send the message right now, due ` +
+          `to operation timeout.`;
+        logger.warning(desc);
+        const e: AmqpError = {
+          condition: ErrorNameConditionMapper.ServiceUnavailableError,
+          description: desc
+        };
+        throw translate(e);
+      }
+      try {
+        this._sender!.sendTimeoutInSeconds = (timeoutInMs - timeTakenByInit) / 1000;
+        const delivery = await this._sender!.send(rheaMessage, undefined, 0x80013700, {
+          abortSignal
```

Comment:
Well sure, tackle #13504 for event hubs while you're at it :smile:

Reply:
I HAD to do it otherwise the test failed!

Reply:
Hmmm, what other tests can I suggest then...